atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+
+            self.parse_thermodynamics_step(data)
+            sec_calc = sec_run.calculation[-1]
+            if format_times(sec_calc.time) != frame:  # TODO check this comparison
+                sec_calc = sec_run.m_create(Calculation)
+                sec_calc.time = frame * time_unit
+            for calc_entry in data_h5md['x_h5md_custom_calculations']:
+                sec_calc.x_h5md_custom_calculations.append(calc_entry)
+            sec_energy = sec_calc.energy
+            if not sec_energy:
+                sec_energy = sec_calc.m_create(Energy)
+            for energy_entry in data_h5md['x_h5md_energy_contributions']:
+                sec_energy.x_h5md_energy_contributions.append(energy_entry)
+
+    def parse_system(self):
+        sec_run = self.archive.run[-1]
+
+        system_info = self.system_info.get('system')
+        if not system_info:
+            self.logger.error('No particle information found in H5MD file.')
+
+        n_frames = len(system_info.get('times', []))
+        self._system_time_map = {}
+        self._system_step_map = {}
+        for frame in range(n_frames):
+            # if (n % self.frame_rate) > 0:
+            #     continue
+
+            atoms_dict = {}
+            for key in system_info.keys():
+                if key == 'times':
+                    time = system_info.get('times', [None] * (frame + 1))[frame]
+                    if time is not None:
+                        self._system_time_map[round(ureg.convert(
+                            time.magnitude, time.units, ureg.picosecond), 5)] = len(self._system_time_map)
+                elif key == 'steps':
+                    step = system_info.get('steps', [None] * (frame + 1))[frame]
+                    if step is not None:
+                        self._system_step_map[round(step)] = len(self._system_step_map)
+                else:
+                    atoms_dict[key] = system_info.get(key, [None] * (frame + 1))[frame]
+
+            if frame == 0:  # TODO extend to time-dependent bond lists
+                connectivity = self.h5md_groups['connectivity']
+                atoms_dict['bond_list'] = self._data_parser.hdf5_getter(connectivity, 'bonds')
+
+            self.parse_trajectory_step({
+                'atoms': atoms_dict
+            })
+
+            if frame == 0:  # TODO extend to time-dependent topologies
+                topology = self._data_parser.hdf5_getter(connectivity, 'particles_group', None)
+                if topology:
+                    self.get_atomsgroup_fromh5md(sec_run.system[frame], topology)
+
+    def parse_method(self):
+
+        sec_method = self.archive.run[-1].m_create(Method)
+        sec_force_field = sec_method.m_create(ForceField)
+        sec_model = sec_force_field.m_create(Model)
+
+        # get the atom parameters
+        n_atoms = self.n_atoms[0]  # TODO Extend to non-static n_atoms
```

Comment:
maybe error handling here if not self.n_atoms?

Reply:
done.